lib/ASN1/Base128.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace FG\ASN1;
+
+/**
+ * A base-128 decoder.
+ */
+class Base128
+{
+    /**
+     * @param integer $value
+     *
+     * @return string
+     */
+    public static function encode($value)
+    {
+        $octets = chr($value & 0x7F);
+        $value >>= 7;
+
+        while ($value > 0) {
+            $octets .= chr(0x80 | ($value & 0x7F));
+            $value >>= 7;
+        }
+
+        return strrev($octets);
```

Comment:
@FGrosse Yeah, I ran into similar issues, until I read [this article](http://www.strozhevsky.com/free_docs/asn1_in_simple_words.pdf) (p3). So I connected the dots and figured both long identifiers and SID's are encoded the same way (base-128). And your second encoding seems the correct one to me as well.